src/umpire/ResourceManager.cpp

```diff
@@ -359,6 +359,23 @@ util::AllocationRecord ResourceManager::deregisterAllocation(void* ptr)
   return m_allocations.remove(ptr);
 }
 
+void* ResourceManager::transfer(void* ptr, Allocator to)
+{
+  auto alloc_record = findAllocationRecord(ptr);
+
+  if (!(alloc_record->strategy->treatAsPassthrough() && to.getAllocationStrategy()->treatAsPassthrough())) {
```

Comment:
So this does a direct transfer when the underlying allocators are the same, but moves when they are different? Wouldn't this allow a direct transfer between named allocators built on different base allocators? Should this take a resource to use for the copy?